packages/web-components/src/components/Modal/Gallery.tsx

```diff
@@ -0,0 +1,26 @@
+import type { AttachmentTestStepResult } from "@allurereport/core-api";
+import { attachmentType } from "@allurereport/web-commons";
+import type { FunctionalComponent } from "preact";
+import { Attachment } from "@/components/Attachment";
+import ModalThumb from "@/components/Modal/Thumb";
+import type { ModalGalleryProps } from "@/components/Modal/index";
+import styles from "./styles.scss";
+
+const Gallery: FunctionalComponent<ModalGalleryProps> = ({ attachments = [] }) => {
+  const filteredAttachments = attachments?.filter(({ link: { contentType } }: AttachmentTestStepResult) => {
+    const type = attachmentType(contentType as string).type;
+
+    return !["archive", null].includes(type as string);
+  });
+
+  return (
+    <div className={styles["modal-gallery"]}>
+      {filteredAttachments?.map((item, index) => (
+        <ModalThumb item={item} key={index}>
```

Comment:
Using the index as a key isn't a good idea. Find something unique in the item (e.g. `uuid`) and use the property as the key